examples/logger/src/bring_your_own_formatter.py

```diff
@@ -1,12 +1,13 @@
 from aws_lambda_powertools import Logger
 from aws_lambda_powertools.logging.formatter import LambdaPowertoolsFormatter
+from aws_lambda_powertools.logging.types import TypedLog
 
 
 class CustomFormatter(LambdaPowertoolsFormatter):
-    def serialize(self, log: dict) -> str:
+    def serialize(self, log: TypedLog) -> str:  # type: ignore
         """Serialize final structured log dict to JSON str"""
-        log["event"] = log.pop("message")  # rename message key to event
-        return self.json_serializer(log)  # use configured json serializer
+        log["event"] = log.pop("message")  # type: ignore
+        return self.json_serializer(log)  # type: ignore
```

Comment:
Check if you still need this after modifying the class.. `type: ignore` is a last resort, we should avoid it. If truly necessary, you can make it more specific like based on Mypy error, e.g., `type: ignore[override]`

Reply:
The issue mypy doesn't like here is two-fold:
- The addition of a key not defined in the TypedDict (in this case "event") raises a "typeddict-unknown-key" error
- Attempting to delete the "message" key from the TypedDict raises a "misc" error because it is a required field (since total=False was removed)

Reply:
Thanks a lot @erikayao93 for explaining it. Apparently we're not the only ones - known issue here: https://github.com/python/mypy/issues/4976

I'd guess a `Dict[str, Any]` would address the first issue, while the former might need a specific `type: ignore[typeddict-unknown-key]`.

Let me know what alternatives you can find, and I can jump in early next week.

Reply:
I did some local testing, away from the repo, and it looks like even with the `TypeAlias`, mypy will still give a `[typeddict-unknown-key]` error when trying to add a key. 

I tried out defining some kind of `Dict[str, Any]` with a `TypedDict` attribute, which actually does work. 

**Pros:**
- elements can be added AND deleted without raising errors

**Cons:**
- the type checking is gone after the initial `TypedDict` definition

I pretty much casted the `TypedDict` to a regular `Dict[str, Any]`, but I don't know if that's what we really want? It mght not be particularly feasible for our use case, especially since any typing is gone after the initial definition.

I attached my small test file here, which compiles without complaint from `mypy` and ran on Python 3.10.6.
[typedict_test.zip](https://github.com/awslabs/aws-lambda-powertools-python/files/11716892/typedict_test.zip)

Let me know your thoughts!

Reply:
I missed this earlier, since I had ended up using a single `type: ignore`, but that ignore is actually muting four errors now -- two new errors came up because of using Mapping
Since the mapping is read-only (which is what makes it compatible with TypedDict), there is an `index` error and a `union-attr` error on that line.
So overall on that line we have:
- error: Unsupported target for indexed assignment ("Union[PowertoolsLogRecord, Mapping[Any, Any]]")  [index]
- error: TypedDict "PowertoolsLogRecord" has no key "event"  [typeddict-unknown-key]
- error: Item "Mapping[Any, Any]" of "Union[PowertoolsLogRecord, Mapping[Any, Any]]" has no attribute "pop"  [union-attr]
- error: Key "message" of TypedDict "PowertoolsLogRecord" cannot be deleted  [misc]

Let me know how you want me to proceed on this!

Reply:
just pushed the changes to fix the first one (`unsupported target for indexed assignment`) and ignored the last two checks with an explanation as to why we did it.

We wouldn't want to add `message` as NotRequired because it wouldn't be accurate (it's always there) - I replied to that thread with context. It's the same as saying `logger.info()` is a valid API call instead of `logger.info("any message here")`.